src/lib/index.js

```diff
@@ -1244,3 +1244,20 @@ lib.ensureUniformFontSize = function(gd, baseFont) {
     );
     return out;
 };
+
+/**
+ * provide a human-readable lists with an optional ending separator such as A, B, C and D
+ *
+ * @param {array} arr : the array to join
+ * @param {string} mainSeparator : main separator
+ * @param {string} lastSeparator : optional last separator
+ *
+ * @return {string} : joined list
+ */
+lib.join2 = function(arr, mainSeparator, lastSeparator) {
+    var len = arr.length;
+    if(len > 2 && lastSeparator) {
```

Comment:
```suggestion
    if(len > 1 && lastSeparator) {
```
If there's only one separator, it's still the last one, right?

Reply:
```suggestion
    if(len > 1) {
```
Or in fact if `lastSeparator isn't optional, as I suggested above ;)
The `else` case is still important for length 1 or 0 though.